lib/dataset/bin.cpp

```diff
@@ -31,8 +31,8 @@ namespace scipp::dataset {
 namespace {
 
 template <class T, class Builder>
-auto bin(const Variable &data, const Variable &indices,
-         const Builder &builder) {
+auto setup_and_apply(const Variable &data, const Variable &indices,
```

Comment:
No change in this file apart from function rename (there as a different function of the same name, and I ran into this the Nth time when considering a solution for the issue using a different approach).